src/storage/FileResourceStore.ts

```diff
@@ -60,16 +62,20 @@ export class FileResourceStore implements ResourceStore {
 
     // Get the path from the request URI, all metadata triples if any, and the Slug and Link header values.
     const path = this.parseIdentifier(container);
-    const { slug, raw } = representation.metadata;
-    const linkTypes = representation.metadata.linkRel?.type;
+    const slug = representation.metadata.get(SLUG)?.value;
+    const type = representation.metadata.get(TYPE)?.value;
+
+    // Create a new container or resource in the parent container with a specific name based on the incoming headers.
+    const isContainer = this.interactionController.isContainer(slug, type);
+    const newIdentifier = this.interactionController.generateIdentifier(isContainer, slug);
+
     let metadata;
+    // eslint-disable-next-line no-param-reassign
```

Comment:
Needed?

Reply:
Yes the linter doesn't allow reassigning properties of function parameters by default.

Reply:
Ah come on! 😆